File: apps/web/src/types/source.ts
import type { CategoryLabel, IsoDateTime, ProcessingStatus, SourceFileType, TeamLabel } from './common'

export type SourceData = {
  id: number
  workspaceId: number
  title: string
  teamLabel: TeamLabel
  categoryLabels: CategoryLabel[]
  fileType: SourceFileType
  originalFilename: string
  storagePath?: string
  periodStartMonth?: string | null
  periodEndMonth?: string | null
  periodLabel?: string | null
  processingStatus: ProcessingStatus
  processingError?: string | null
  uploadedAt: IsoDateTime
  processedAt?: IsoDateTime | null
  deletedAt?: IsoDateTime | null
}

export type CreateSourceInput = {
  workspaceId: number
  title: string
  file: File
  teamLabel: TeamLabel
  categoryLabels: CategoryLabel[]
  periodStartMonth?: string
  periodEndMonth?: string
}

export type SourceDataApi = {
  id: number
  workspace_id: number
  title: string
  team_label: TeamLabel
  category_labels: CategoryLabel[]
  file_type: SourceFileType
  original_filename: string
  storage_path: string
  period_start_month?: string | null
  period_end_month?: string | null
  period_label?: string | null
  processing_status: ProcessingStatus
  processing_error?: string | null
  uploaded_at: IsoDateTime
  processed_at?: IsoDateTime | null
  deleted_at?: IsoDateTime | null
}

export type SourceSummary = {
  total: number
  processing: number
  ready: number
  failed: number
}

File: apps/api/app/schemas/source.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import CategoryLabel, ProcessingStatus, SourceFileType, TeamLabel


class SourceCreate(BaseModel):
    workspace_id: int
    title: str
    team_label: TeamLabel
    category_labels: list[CategoryLabel]
    file_type: SourceFileType
    original_filename: str
    period_start: datetime | None = None
    period_end: datetime | None = None
    period_label: str | None = None


class SourceRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workspace_id: int
    title: str
    team_label: TeamLabel
    category_labels: list[CategoryLabel]
    file_type: SourceFileType
    processing_status: ProcessingStatus
    processing_error: str | None = None
    original_filename: str
    storage_path: str
    period_start: datetime | None = None
    period_end: datetime | None = None
    period_label: str | None = None
    uploaded_at: datetime
    processed_at: datetime | None = None
    deleted_at: datetime | None = None
